### PR TITLE
Added deploy file and restructured cloud function directory

### PR DESCRIPTION
**Pull request by kabeer27**
Added a deploy.sh file for deploying cloud function sync, takes care of cases when the scheduler job/pub sub already exists.
Symlinked the requirements.txt

@oliverchang

### CODE REVIEW

File: infra/build/functions/sync/sync_test.py [hunk 1]
```diff
@@ -23,10 +23,10 @@
 
 from google.cloud import ndb
```
Comment (oliverchang) [inline infra/build/functions/sync/sync_test.py:25]: Rename this file main_test.py now too.

File: infra/build/functions/sync/deploy.sh [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+# limitations under the License.
+#
+################################################################################
+
+JOB_TOPIC=sync-topic
+SCHEDULER_JOB=sync-scheduler
+JOB_SCHEDULE="*/30 * * * *"
+MESSAGE="Start Sync"
+ENTRY_POINT=sync
+
```
Comment (oliverchang) [inline infra/build/functions/sync/deploy.sh:22]: You need to add an argument for the Cloud Project such that someone can do `./deploy.sh my-cloud-project`.

File: infra/build/functions/sync/deploy.sh [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+
+JOB_TOPIC=sync-topic
```
Comment (oliverchang) [inline infra/build/functions/sync/deploy.sh:17]: Please rename this to schedule-project-sync

File: infra/build/functions/sync/deploy.sh (hunk 2 above)
Comment (kabeer27) [reply infra/build/functions/sync/deploy.sh:22]: Should i make it an optional or a required argument

File: infra/build/functions/sync/deploy.sh [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+################################################################################
+
+JOB_TOPIC=schedule-project-sync
+SCHEDULER_JOB=sync-scheduler
+JOB_SCHEDULE="*/30 * * * *"
+MESSAGE="Start Sync"
+ENTRY_POINT=sync
+
+if [ "$1" ]; then
+	gcloud config set project $1
```
Comment (oliverchang) [inline infra/build/functions/sync/deploy.sh:24]: This sets the project globally for the user, which is not desired. Instead, please pass --project to all gcloud commands.

**Pull request merged**